Adapter.C

```diff
@@ -923,6 +925,53 @@ void preciceAdapter::Adapter::setupCheckpointing()
     ACCUMULATE_TIMER(timeInCheckpointingSetup_);
 }
 
+void preciceAdapter::Adapter::pruneCheckpointedFields()
+{
+    // Check if checkpointed fields exist in OpenFOAM database
+    // If not, remove them from the checkpointed fields list
+
+    word obj;
+    std::vector<word> fieldNames;
+    std::vector<std::string> toRemove;
+#undef doLocalCode
+#define doLocalCode(GeomField, GeomFieldMap, GeomFieldCopiesMap)                                       \
```

Comment:
I think it is fine having this as a macro for now. We can consider refactoring later, in a broader context.